### PR TITLE
[2.19.x] DDF-6232 decrypt password parameter in SmtpClientImpl

### PR DESCRIPTION
**Pull request by rymach**
If a SmtpClientImpl config file is provided a bug occurs where the password parameter is sent to the smtp server still encrypted. Note: if no configuration file is provided and the configuration is instead performed through the admin ui - everything works properly. This change decrypts the password parameter in SmtpClientImpl when necessary. 

#### How should this be tested?

- create config file: org.codice.ddf.platform.email.impl.SmtpClientImpl.config 
with a smtp server url, port, username, and password 
- cause ddf to send an email. one way to do this is to check "Enable Query Feedback" in Admin UI > Catalog UI > Catalog UI Search; and supply a "Query Feedback Email Destination"

#### What are the relevant tickets?
Fixes: #6232

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: platform/email/platform-email-impl/src/main/resources/OSGI-INF/blueprint/blueprint.xml [hunk 1]
```diff
@@ -18,7 +18,12 @@
            xsi:schemaLocation="http://www.osgi.org/xmlns/blueprint/v1.0.0
            http://www.osgi.org/xmlns/blueprint/v1.0.0/blueprint.xsd">
 
+    <reference id="encryptionService"
+        interface="ddf.security.encryption.EncryptionService"
+        availability="mandatory"/>
```
Comment (bdeining) [inline platform/email/platform-email-impl/src/main/resources/OSGI-INF/blueprint/blueprint.xml:23]: i think this is default

Comment (rymach) [reply platform/email/platform-email-impl/src/main/resources/OSGI-INF/blueprint/blueprint.xml:23]: image

Comment (rymach) [reply platform/email/platform-email-impl/src/main/resources/OSGI-INF/blueprint/blueprint.xml:23]: I'm not sure if it's necessary or not - I just copy pasted from another location in the codebase

**Comment by rymach** [comment]
todo: forwardport after this is merged

File: platform/email/platform-email-impl/src/main/java/org/codice/ddf/platform/email/impl/SmtpClientImpl.java [hunk 2]
```diff
@@ -92,7 +99,7 @@ public void setUserName(@Nullable String userName) {
    * @param password the password for the email server
    */
   public void setPassword(@Nullable String password) {
-    this.password = password;
+    this.password = encryptionService.decryptValue(password);
```
Comment (Lambeaux) [inline platform/email/platform-email-impl/src/main/java/org/codice/ddf/platform/email/impl/SmtpClientImpl.java:102]: ❓ Do we want to keep the password encrypted in-memory and only decrypt on use?

Comment (rymach) [reply platform/email/platform-email-impl/src/main/java/org/codice/ddf/platform/email/impl/SmtpClientImpl.java:102]: I assumed that's how this worked already - is it not?

**Comment by bdeining** [comment]
build now

File: platform/email/platform-email-impl/src/main/java/org/codice/ddf/platform/email/impl/SmtpClientImpl.java (hunk 2 above)
Comment (bdeining) [reply platform/email/platform-email-impl/src/main/java/org/codice/ddf/platform/email/impl/SmtpClientImpl.java:102]: why? isn't that actually a safer approach?

Comment (rymach) [reply platform/email/platform-email-impl/src/main/java/org/codice/ddf/platform/email/impl/SmtpClientImpl.java:102]: where does the decryption key live

Comment (shaundmorris) [reply platform/email/platform-email-impl/src/main/java/org/codice/ddf/platform/email/impl/SmtpClientImpl.java:102]: Someone from the @codice/security team could probably help. Is it only accessible here? https://github.com/codice/ddf/blob/2.19.x/platform/email/platform-email-impl/src/main/java/org/codice/ddf/platform/email/impl/SmtpClientImpl.java#L162
If so then possible could decrypt it there so it isn't in memory. Not sure if that is better.

Comment (pklinef) [reply platform/email/platform-email-impl/src/main/java/org/codice/ddf/platform/email/impl/SmtpClientImpl.java:102]: The decryption key lives in `etc/ketsets` which is protected by the security policy.

https://github.com/codice/ddf/blob/master/distribution/ddf-common/src/main/resources/security/default.policy#L234-L235

**Comment by rymach** [comment]
build now

**Comment by rymach** [comment]
build now

**Pull request merged**